src/engine/controls/cuecontrol.cpp

```diff
@@ -1160,29 +1175,27 @@ void CueControl::hotcueEndPositionChanged(
 }
 
 void CueControl::hintReader(HintVector* pHintList) {
-    Hint cueHint;
-    const auto mainCuePosition =
-            mixxx::audio::FramePos::fromEngineSamplePosMaybeInvalid(
-                    m_pCuePoint->get());
-    if (mainCuePosition.isValid()) {
-        cueHint.frame = static_cast<SINT>(mainCuePosition.toLowerFrameBoundary().value());
-        cueHint.frameCount = Hint::kFrameCountForward;
-        cueHint.priority = 10;
-        pHintList->append(cueHint);
-    }
+    appendCueHint(pHintList, m_pCuePoint->get(), Hint::Type::MainCue);
 
     // this is called from the engine thread
     // it is no locking required, because m_hotcueControl is filled during the
     // constructor and getPosition()->get() is a ControlObject
     for (const auto& pControl : qAsConst(m_hotcueControls)) {
-        const mixxx::audio::FramePos position = pControl->getPosition();
-        if (position.isValid()) {
-            cueHint.frame = static_cast<SINT>(position.toLowerFrameBoundary().value());
-            cueHint.frameCount = Hint::kFrameCountForward;
-            cueHint.priority = 10;
-            pHintList->append(cueHint);
+        appendCueHint(pHintList, pControl->getPosition(), Hint::Type::HotCue);
+    }
+
+    CuePointer pAudibleSound =
+            m_pLoadedTrack->findCueByType(mixxx::CueType::AudibleSound);
+    if (pAudibleSound) {
+        const mixxx::audio::FramePos frame = pAudibleSound->getPosition();
+        if (frame.isValid()) {
+            appendCueHint(pHintList, frame, Hint::Type::FirstSound);
         }
```

Comment:
`frame` is already checked for validity inside the `appendCueHint`. What's the protocol here, trust the caller and don't check or trust the callee to check? Neither?

Reply:
I have removed the redundant isValid() check and amend it to the last commit.